File: exercism/kotlin/sieve/src/main/kotlin/Sieve.kt
object Sieve {
  fun primesUpTo(limit: Int): List<Int> {
    val prime_candidate = Array(limit + 1, { true })
    return (2 .. limit).filter {
      if (prime_candidate[it]) {
        for (k in it .. limit / it) {
          prime_candidate[k * it] = false
        }
      }
      prime_candidate[it]
    }
  }
}

File: exercism/ruby/sieve/sieve.rb
module BookKeeping
  VERSION = 1
end

class Sieve
  attr_reader :primes

  def initialize(limit)
    sieve = Array.new(limit + 1, true)
    @primes = []

    2.upto(limit) do |candidate|
      if sieve[candidate]
        @primes << candidate
        candidate.step(limit, candidate) do |multiple|
          sieve[multiple] = false
        end
      end
    end
  end
end
